packages/bitcore-node/src/routes/api/address.ts

```diff
@@ -30,6 +31,30 @@ router.get('/:address', function(req, res) {
   ChainStateProvider.streamAddressUtxos(payload);
 });
 
+router.get('/:address/coins', async function (req, res) {
+  let { address, chain, network } = req.params;
+
+  try {
+    let coins = await CoinStorage.collection.find({ address, chain, network }).toArray(); //rename coins
+    let spentTxids = coins.filter((tx) => { return tx.spentTxid; }).map((tx) => tx.spentTxid);
+    let mintedTxids = coins.filter((tx) => { return tx.mintTxid; }).map((tx) => tx.mintTxid);;
+
+    let fundingTxInputs = await CoinStorage.collection.find({ chain, network, spentTxid: { $in: mintedTxids } }).toArray();
+    let fundingTxOutputs = await CoinStorage.collection.find({ chain, network, mintTxid: { $in: mintedTxids } }).toArray();
+
+    let spendingTxInputs: any = [];
+    let spendingTxOutputs: any = [];
+
+    if (!(spentTxids === null)) {
```

Comment:
if(spentTxids)

Reply:
let's batch these queries into a Promise.all. and default spentTxids to empty array, and remove this if check
```
let [fundingTxInputs, fundingTxOutputs, spendingTxInputs, spendingTxOutputs] = await Promise.all([
  query,
  query,
  query...
])
```